cloud/disk_manager/internal/pkg/auth/credentials.go

```diff
@@ -26,9 +28,27 @@ func NewCredentials(
 	if len(config.GetMetadataUrl()) == 0 {
 		return nil
 	}
-
-	return &credentialsWrapper{
-		impl: metadata.NewInstanceServiceAccount(
+	var impl Credentials
+	if config.GetServiceAccount() != nil {
+		_impl, err := credentials.NewOauth2TokenExchangeCredentials(
+			credentials.WithTokenEndpoint(config.GetMetadataUrl()),
+			credentials.WithAudience(*config.ServiceAccount.Audience),
+			credentials.WithJWTSubjectToken(
+				credentials.WithSigningMethod(jwt.SigningMethodRS256),
+				credentials.WithKeyID(*config.ServiceAccount.KeyId),
+				credentials.WithRSAPrivateKeyPEMFile(config.GetCertFile()),
+				credentials.WithIssuer(*config.ServiceAccount.Id),
+				credentials.WithSubject(*config.ServiceAccount.Id),
+				credentials.WithAudience(*config.ServiceAccount.Audience),
+			),
+		)
+		if err != nil {
+			logging.Error(ctx, "failed to create token credential, error: %s", err)
```

Comment:
failed to create token credentials

Reply:
./credentials.go:50:4: github.com/ydb-platform/nbs/cloud/tasks/logging.Error does not support error-wrapping directive %w
I would stick with %s